src/analysis/ControlFlowGraph.h

```diff
@@ -127,13 +135,23 @@ namespace vc4c
 
     struct LoopInclusionTreeNodeBase
     {
-        LoopInclusionTreeNodeBase* findRoot();
+        virtual ~LoopInclusionTreeNodeBase() = default;
+
+        LoopInclusionTreeNodeBase* findRoot(Optional<int> depth);
+        unsigned int longestPathLengthToRoot() const;
+        bool hasCFGNodeInChildren(const CFGNode* node) const;
+
+        std::string dumpLabel() const;
     };
 
     using LoopInclusionTreeNode =
         Node<ControlFlowLoop*, LoopInclusion, Directionality::DIRECTED, LoopInclusionTreeNodeBase>;
     using LoopInclusionTreeEdge = LoopInclusionTreeNode::EdgeType;
 
+    // Cannot expose these functions due to link error.(?)
```

Comment:
Do you need to expose those functions outside of the ControlFlowGraph.cpp file?

Reply:
Yes. I wrote the same code snippet at https://github.com/doe300/VC4C/blob/4dfd55523ec295479aaf64cfe4d48ada4b224337/src/optimization/ControlFlow.cpp#L1277-L1281. So I want to combine those codes.

Reply:
And why does the linker not like the code?